R/tt_sort.R

```diff
@@ -50,29 +57,105 @@ cont_n_onecol <- function(j) {
+#'   at a particular Path in the table tree.
+#' 
 #' @inheritParams gen_args
 #' @param scorefun function. Scoring function, should accept the type of
-#'   children directly under the position at \code{path} (either VTableTree,
-#'   VTableRow, or VTableNodeInfo, which covers both) and return a numeric value
+#'   children directly under the position at \code{path} (either `VTableTree`,
+#'   `VTableRow`, or `VTableNodeInfo`, which covers both) and return a numeric value
 #'   to be sorted.
 #' @param decreasing logical(1). Should the the scores generated by
 #'   \code{scorefun} be sorted in decreasing order. If unset (the default of
 #'   \code{NA}), it is set to \code{TRUE} if the generated scores are numeric
 #'   and \code{FALSE} if they are characters.
-#' @param na.pos character(1). What should be done with children (subtrees/rows)
+#' @param na.pos character(1). What should be done with children (sub-trees/rows)
 #'   with \code{NA} scores. Defaults to \code{"omit"}, which removes them, other
 #'   allowed values are \code{"last"}  and \code{"first"} which indicate where
 #'   they should be placed in the order.
 #' @param .prev_path character. Internal detail, do not set manually.
-#' @return A TableTree with the same structure as \code{tt} with the exception
+#' 
+#' @return A `TableTree` with the same structure as \code{tt} with the exception
 #'   that the requested sorting has been done at \code{path}
-#' @details The \code{path} here can include \code{"*"} as a step, which means
-#' that each child at that step will be \emph{separately} sorted based on
-#' \code{scorefun} and the remaining \code{path} entries. This can occur
-#' multiple times in a path.
-#'
-#'
+#'   
+#' @details The \code{path} here can include the "wildcard" \code{"*"} as a step, 
+#'   which translates roughly as *any* node/branching element and means
+#'   that each child at that step will be \emph{separately} sorted based on
+#'   \code{scorefun} and the remaining \code{path} entries. This can occur
+#'   multiple times in a path. 
+#'   
+#'   Note that sorting needs a deeper understanding of table structure in 
+#'   `rtables`. Please consider reading related vignette 
+#'   ([Sorting and Pruning](https://insightsengineering.github.io/rtables/main/articles/sorting_pruning.html))
+#'   and explore table structure with useful functions like [table_structure()]
+#'   and [row_paths_summary()]. It is also very important to understand the 
+#'   difference between "content" rows and "data" rows. The first one analyzes
+#'   and describes the split variable generally and is generated with 
+#'   [summarize_row_groups()], while the second one is commonly produced by 
+#'   calling one of the various [analyze()] instances.
+#'   
+#' @seealso Built-in score functions are [cont_n_allcols()] and [cont_n_onecol()].
+#'   They are both working with content rows (coming from [summarize_row_groups()])
```

Comment:
This discussion belongs in the details section, see also should be mostly links, with little-to-no explanatory text in it.

Reply:
It makes sense! Moving sections. Thanks